addons/mail/static/src/components/composer/composer_tests.js

```diff
@@ -274,22 +274,602 @@ QUnit.test('add emoji replaces (keyboard) text selection', async function (asser
     await nextAnimationFrame();
 });
 
-QUnit.test('display partner mention suggestions on typing "@"', async function (assert) {
+QUnit.test('display canned response suggestions on typing ":"', async function (assert) {
+    assert.expect(2);
+
+    this.data['mail.shortcode'].records = [{
```

Comment:
`push` into `records` is necessary for some models (users, partners) so it is preferable to always use it to simplify the guideline. Also it is the equivalent of doing `create` in the server.